admin/projects.go

```diff
@@ -191,3 +194,27 @@ func (s *Service) TriggerReconcile(ctx context.Context, deploymentID string) err
 	}()
 	return nil
 }
+
+func (s *Service) UpdateProject(ctx context.Context, p *database.Project) (*database.Project, error) {
+	// TODO: Make this actually fault tolerant.
+
+	ds, err := s.DB.FindDeployments(ctx, p.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &provisioner.UpdateProvisionOptions{Envs: p.Envs}
+	for _, d := range ds {
+		err := s.provisioner.UpdateProvision(ctx, opts, d.RuntimeHost, d.RuntimeInstanceID)
```

Comment:
The logic in `UpdateProvision` should be implemented directly in the `admin.Service` (even in this file, maybe this func). The provisioner should just be for picking a runtime host and creating/deleting an instance. Subsequent config can be made directly by RPCs to the runtime from `admin.Service`